src/heartbeat/utils.js

```diff
@@ -50,10 +59,19 @@ function assignContextDefaults(ctx, def) {
     def = {};
   }
 
+  const heartbeatTimestamp = res.timestamp;
+  const collectorToken = refocusInstance.token;
+  const secret = collectorToken + heartbeatTimestamp;
+
   Object.keys(def).forEach((key) => {
     if (!ctx.hasOwnProperty(key) && def[key].hasOwnProperty('default')) {
       ctx[key] = def[key].default;
     }
+
+    if (ctx.hasOwnProperty(key) && def.hasOwnProperty(key) &&
+      def[key].encrypted) {
+      ctx[key] = commonUtils.decrypt(ctx[key], secret, encryptionAlgorithm);
```

Comment:
Do we need to handle the case in which decrypt function fails for some reason (eg, incorrect algo) ? @iamigo

Reply:
Can you please create a user story to deal with that?